File: socketd-broker/socketd-broker-websocket/src/main/java/org/noear/socketd/broker/websocket/WsClientConnector.java
package org.noear.socketd.broker.websocket;

import org.noear.socketd.client.ClientChannel;
import org.noear.socketd.client.ClientConnectorBase;
import org.noear.socketd.protocol.Channel;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

/**
 * @author noear
 * @since 2.0
 */
public class WsClientConnector extends ClientConnectorBase<WsClient> {
    private static final Logger log = LoggerFactory.getLogger(WsClientConnector.class);

    private SocketClientImpl real;

    public WsClientConnector(WsClient client){
        super(client);
    }

    @Override
    public Channel connect() throws IOException {
        SocketClientImpl real = new SocketClientImpl(client.uri(),client);
        return new ClientChannel(real.getChannel(), this);
    }

    @Override
    public void close() throws IOException {
        if (real == null) {
            return;
        }

        try {
            real.close();
        } catch (Throwable e) {
            log.debug("{}", e);
        }
    }
}

File: javascript/socketd/socketd_websocket/WsClientConnector.ts
import {ClientConnectorBase} from "../socketd/transport/client/ClientConnector";
import { ChannelInternal } from "../socketd/transport/core/Channel";
import {WsClient} from "./WsClient";

export class WsClientConnector extends ClientConnectorBase<WsClient> {
    constructor(client: WsClient) {
        super(client);
    }

    connect(): ChannelInternal {
        throw new Error("Method not implemented.");
    }

    close() {
        throw new Error("Method not implemented.");
    }
}